temporal/api/cloud/identity/v1/message.proto

```diff
@@ -70,12 +70,24 @@ message User {
     google.protobuf.Timestamp last_modified_time = 8;
 }
 
+message GoogleGroupSpec {
+    // The email address of the Google group.
+    // The email address is immutable. Once set during creation, it cannot be changed.
+    string email_address = 1;
+}
+
 message UserGroupSpec {
-    // The name of the group as defined in the customer's IdP (e.g. Google group name in Google Workspace)
-    // The name is immutable. Once set, it cannot be changed
-    string name = 1;
-    // The access assigned to the group
+    // The display name of the group.
+    string display_name = 1;
+    // The access assigned to the group.
     Access access = 2;
+    // The type of the group.
+    // Currently, the only supported type is 'google-group'.
+    // The type is immutable. Once set during creation, it cannot be changed.
+    string type = 3;
+    // The specification of the google group.
+    // This field is required when the group type is 'google-group'.
+    GoogleGroupSpec google_group = 4;
```

Comment:
Can't the presence of `google_group` be enough to not need `type`? Would you consider a `oneof`? Same for the filter on the request.

Reply:
I think I will sight the discussion here:
https://github.com/temporalio/api-cloud/pull/23#discussion_r1593105415
We decided not to use one-off but rely on presence or absence of fields instead of a explicit type.
Let me remove the type field.